plugins/rebalance/rest.go

```diff
@@ -44,14 +44,23 @@ func rebalanceStartHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	rebalinfo := createRebalanceInfo(volname, &req)
-	if rebalinfo == nil {
-		logger.WithError(err).Error("failed to create Rebalance info")
+	rebalInfo, err := GetRebalanceInfo(volname)
+	if err == nil {
+		if rebalInfo.State == rebalanceapi.Started {
+			log.WithError(err).WithField("volume", volname).Error("Rebalance process has already been started.")
+			restutils.SendHTTPError(ctx, w, http.StatusBadRequest, ErrRebalanceAlreadyStarted)
+			return
+		}
+	}
+
+	rebalInfo = createRebalanceInfo(volname, &req)
+	if rebalInfo == nil {
```

Comment:
question: in what scenario  ```createRebalanceInfo``` will return nil?

Reply:
I think this scenario won't happen, but I have kept the error handling in case of future use.

Reply:
if this is for future use, this check will become redundant now. then we need to comment the code or remove it